packages/flutter_tools/test/commands.shard/hermetic/build_aar_test.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/build_system/build_system.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/build.dart';
+import 'package:unified_analytics/unified_analytics.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+import '../../src/fake_process_manager.dart';
+import '../../src/fakes.dart';
+import '../../src/test_build_system.dart';
+import '../../src/test_flutter_command_runner.dart';
+
+void main() {
+  late BufferLogger logger;
+  late MemoryFileSystem fs;
+  late FakeProcessManager processManager;
+  late Platform platform;
+  late Cache cache;
+  late FakeAnalytics fakeAnalytics;
+
+  setUpAll(() {
+    Cache.disableLocking();
+  });
+
+  setUp(() {
+    fs = MemoryFileSystem.test();
+    final Directory flutterRoot = fs.directory('flutter');
+    Cache.flutterRoot = flutterRoot.path;
+    logger = BufferLogger.test();
+    platform = FakePlatform(environment: const <String, String>{'PATH': ''});
+    processManager = FakeProcessManager.empty();
+    cache = Cache.test(
+      rootOverride: flutterRoot,
+      logger: logger,
+      processManager: processManager,
+    );
+    fakeAnalytics = getInitializedFakeAnalyticsInstance(
+      fs: fs,
+      fakeFlutterVersion: FakeFlutterVersion(),
+    );
+  });
+
+  testUsingContext('will not build an AAR for a plugin', () async {
```

Comment:
If the tested behavior did not already exist on the 3.16 branch, let's just delete this test, per https://github.com/flutter/flutter/pull/141019/files#r1445425246